### DISCUSSION TITLE
Looking for a way to adapt theme colors based on shell

### DISCUSSION THREAD

**Discussion by RustyBrakes** [JanDeDobbeleer/oh-my-posh#1559] [Troubleshoot] [answered]
Hello there, I hope I'm not missing anything obvious, but I've searched for just about every keyword I could think of here and I've reviewed many other themes to look for a similar thing. I've also tried looking through the .../main/src/segment_XXX code to see if I can reverse engineer something but no luck so far. 

**AIM: To have a different background color depending on shell and/or OS**

My attempt at this section so far looks like this:

```
{
  "type": "shell",
  "style": "plain",
  "foreground": "#ffffff",
  "background_templates": [
    "{{if contains \"pwsh\" .Shell}}#0000FF{{end}}",
    "{{if contains \"zsh\" .Shell}}#00FFFF{{end}}",
    "{{if contains \"bash\" .Shell}}#B388FF{{end}}"
  ],
  "properties": {
    "mapped_shell_names": {
      "pwsh": "PS"
    }
  }
}
```

This ain't working and I guess there's no variable that I can test against? (like the .Shell I'm using here)

**Comment by RustyBrakes** [comment, marked as answer]
I've worked out a way to achieve this if anyone else is looking for the same thing! It uses [color override](https://ohmyposh.dev/docs/config-colors#color-overrides) and that's possible for both OS and shell segments because they have name overrides like in the example below. 

I do slightly wish that there was a way to apply logic like "if shell=powershell then hide segment" but perhaps that's not really possible within the framework here. 

Thanks to Jan DeDobbeleer for putting together a great tool!

```
{
"type": "shell",
          "style": "plain",
          "foreground": "#ffffff",
          "background": "transparent",
          "properties": {
            "mapped_shell_names": {
              "pwsh": "<#0273B7>PS \u276F_</>",
              "bash": "<#34024D>bash</>"
        }
    }
}
```

**Reply by JanDeDobbeleer** [reply]
When #1565 lands you should be able to do your idea as well now

**Reply by RustyBrakes** [reply]
Thanks @JanDeDobbeleer ! Does that mean that I can do things like "if shell = bash then path color = orange" or "if exit code gt 0 then OS icon = red" ?? Like the environment info can be grabbed from any segment?

Very cool if so!

**Reply by JanDeDobbeleer** [reply]
Yes, I should add the ExitCode though, that's not yet part of it.

**Reply by JanDeDobbeleer** [reply]
All done.

**Reply by RustyBrakes** [reply]
Thank you @JanDeDobbeleer ! I really don't want to sound demanding, but would it be possible to add .OS as an environmental variable too? I'm picturing a setup where the prompt changes colour based on windows, mac, debian, ubuntu, raspbian etc. 

If I am working via a prompt, I really want to have a quick or natural way to remember which commands to use for that OS, and I feel like setting colour by OS would be a great way to do that!

As I said though, you've already put an update forward that directly solved my issue within 24 hours which is unbelievable. So I don't want to put any pressure on this!

**Reply by JanDeDobbeleer** [reply]
We do have the OS segment? But it's definitely possible.

**Reply by RustyBrakes** [reply]
Picture this though - you have a certain style of OMP theme which will change from a blue colour theme (Windows) to Ubuntu's orange/purple, or to the raspberry pi colours for raspbian, or any other distribution. 

In the two examples below, I can set the shell for my colour scheme, and I use the shell segment again for the final character of the prompt, but I cannot adjust based on OS, which means I would only have colours based on shell:

![powershell](https://user-images.githubusercontent.com/6842623/149593043-663194a4-3ffa-4d5f-ac73-a4a0c0d375c6.png)
![wsl ubuntu](https://user-images.githubusercontent.com/6842623/149593045-2af6d36b-753b-461c-b1bd-df754c96eb52.png)

**Reply by RustyBrakes** [reply]
This is the exact same theme json, but dynamically changing based on shell. Ideally I'd prefer to dynamically change based on .OS as well as .Shell. The result would be a single theme I could use everywhere, with all of the beautiful features, but still making it very clear which OS I am running on

**Reply by RustyBrakes** [reply]
I'm sorry, I feel guilty I didn't review the changes you were making in #1565 and mention the OS idea before you'd made your merge!

**Reply by JanDeDobbeleer** [reply]
I'm too fast anyways. Will add it.
